File: backend/app/schemas/ai.py
from datetime import datetime
from typing import List, Optional, Dict, Any
from uuid import UUID
from pydantic import BaseModel, ConfigDict

class AIMessageResponse(BaseModel):
    id: UUID
    conversation_id: UUID
    role: str
    content: str
    confidence_score: Optional[str]
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class AIConversationResponse(BaseModel):
    id: UUID
    session_id: UUID
    title: str
    current_goal: Optional[str]
    completed_tasks: Optional[Dict[str, Any]]
    pending_tasks: Optional[Dict[str, Any]]
    active_entities: Optional[Dict[str, Any]]
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class AISessionResponse(BaseModel):
    id: UUID
    user_id: UUID
    status: str
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class AIChatRequest(BaseModel):
    conversation_id: UUID
    prompt: str

class AIChatResponse(BaseModel):
    content: str
    confidence: str = "HIGH"
    conversation_id: UUID

class AgentExecutionResponse(BaseModel):
    id: UUID
    agent_id: UUID
    tool_used: Optional[str]
    duration_ms: int
    inputs_payload: Optional[Dict[str, Any]]
    outputs_payload: Optional[Dict[str, Any]]
    confidence: str
    error_message: Optional[str]
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class ToolDefinitionResponse(BaseModel):
    id: UUID
    name: str
    description: str
    input_schema: Dict[str, Any]
    required_permissions: str
    timeout_seconds: int
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/ai.ts
export interface LLMProvider {
  id: string;
  name: string;
  code: string;
  api_endpoint?: string;
  status: string;
}

export interface ModelConfiguration {
  id: string;
  provider_id: string;
  model_name: string;
  temperature: number;
  max_tokens: number;
  is_default: boolean;
}

export interface AISession {
  id: string;
  user_id: string;
  status: string;
}

export interface AIConversation {
  id: string;
  session_id: string;
  title: string;
  current_goal?: string;
  completed_tasks?: Record<string, any>;
  pending_tasks?: Record<string, any>;
  active_entities?: Record<string, any>;
}

export interface AIMessage {
  id: string;
  conversation_id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  confidence_score?: 'HIGH' | 'MEDIUM' | 'LOW';
  created_at: string;
}

export interface AgentDefinition {
  id: string;
  name: string;
  code: string;
  capabilities: string;
  priority: number;
  version: number;
}

export interface AgentExecution {
  id: string;
  agent_id: string;
  tool_used?: string;
  duration_ms: number;
  inputs_payload?: Record<string, any>;
  outputs_payload?: Record<string, any>;
  confidence: 'HIGH' | 'MEDIUM' | 'LOW';
  error_message?: string;
  created_at: string;
}

export interface ToolDefinition {
  id: string;
  name: string;
  description: string;
  input_schema: Record<string, any>;
  required_permissions: string;
  timeout_seconds: number;
}
